src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,62 @@
 package core.basesyntax;
 
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+
 public class WorkWithFile {
+    private static final int ARRAY_VALUE_INDEX = 1;
+    private static final int ARRAY_NAME_INDEX = 0;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String data = readFile(fromFileName);
+        String report = generateReport(data);
+        writeToFile(toFileName, report);
+    }
+
+    private String readFile(String fromFileName) {
+        File fileCopy = new File(fromFileName);
+        StringBuilder stringBuilder = new StringBuilder();
+
+        try (Scanner scanner = new Scanner(fileCopy)) {
+            while (scanner.hasNext()) {
+                stringBuilder.append(scanner.next()).append(System.lineSeparator());
+            }
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException(e);
```

Comment:
It is good practice to provide a more informative message when re-throwing exceptions. For example: 'throw new RuntimeException("Cannot read file: " + fromFileName, e);'

Reply:
It is good practice to provide a more informative message when re-throwing exceptions. For example: 'throw new RuntimeException("Cannot read file: " + fromFileName, e);'